### ISSUE TITLE
homebrew_2

### ISSUE THREAD

**Issue by perasmus** [jide/clamp#5]
**TL;DR**
Until this formula gets pubished via homebrew, installing it is easy none the less. To give it a try open Terminal and copy
`brew install https://raw.githubusercontent.com/perasmus/clamp_formula/master/clamp.rb`

done. clamp is now installed and hooked up.

**Long explanation**
Hi again,

so I did some tests and tried to come up with the easiest solution possible. Here it is `clamp.rb`:

``` ruby
require "formula"

class Clamp < Formula
  homepage "http://jide.github.io/clamp"
  url "https://github.com/jide/clamp.git"
  version "0.1"

  depends_on "mariadb"

  def install
    inreplace "clamp", /\/usr\/local\/clamp/, prefix
    prefix.install Dir["*"]
    bin.install_symlink '../clamp'
  end

  test do
    system bin/"clamp", "help"
  end
end

```

This formula ist working with jide's current repository. So how does it work?
1. it needs a homepage url.
2. it needs a download url. using the git address doesn't require an explicit checksum.
3. we need a manually added version number. those need to be manually updated.
4 we let homebrew know that we depend on mariadb.
5. these are the installation instructions. homebrew downloads the repo in a temp folder. here we can edit them.
6. we need to replace the absolute path instructions `/usr/local/clamp` in the `clamp` file with whatever homebrew wants to use. this path luckily is saved in the `prefix` variable.
7. we then simply copy everything from the temp folder into the production folder.
8. last, we hookup a symlink from homebrews local `bin` folder to our `clamp` file. homebrew will then again symlink this `bin` file to its bin collection, so that it is globally accessible.
9. this is a small test. if everything is hooked up correctly it will simply pass. if any error would be encountered it would throw an exception.

You can give it a try by putting this `clamp.rb` file into `$(brew --prefix)/Library/Formula/`.
this formula passes `brew audit clamp`, `brew test clamp` and `brew install clamp`. It doesn't require any changes to the current repo. Thus it remains »backwards compatibility«.
